spot_ros2_control/hardware/include/spot_ros2_control/spot_joint_map.hpp

```diff
@@ -50,64 +61,64 @@ static const std::unordered_map<std::string, size_t> kJointNameToIndexWithoutArm
     {"rear_left_knee", 8},    {"rear_right_hip_x", 9}, {"rear_right_hip_y", 10}, {"rear_right_knee", 11},
 };
 
-// TODO(tcappellari): Find a cleaner + better way to load and change these
-// Gain values https://github.com/boston-dynamics/spot-cpp-sdk/blob/master/cpp/examples/joint_control/constants.hpp
-// NOTE: these should be different depending on number of joints the robot has (arm or none)
-// Right now this is just temporary to see if we can get the commands accepted by robot
-// This could be handled via a parameter in the future.
-// kp and kd gains for a robot without an arm
-static const std::vector<float> no_arm_kp = {624, 936, 286, 624, 936, 286, 624, 936, 286, 624, 936, 286};
-static const std::vector<float> no_arm_kd = {5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20, 5.20, 2.04};
-// kp and kd gains for a robot with an arm
-static const std::vector<float> arm_kp = {624, 936, 286,  624, 936, 286, 624, 936, 286, 624,
-                                          936, 286, 1020, 255, 204, 102, 102, 102, 16.0};
-static const std::vector<float> arm_kd = {5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20,
-                                          5.20, 2.04, 10.2, 15.3, 10.2, 2.04, 2.04, 2.04, 0.32};
+/// @brief Return the joint name to index map depending on the namespace and if the robot has an arm.
+/// @param spot_name Namespace that the ros2 control stack was launched in that prefixes the joint names
+/// @param has_arm Boolean indicating if the arm joint angles should be included in the map
+/// @return Unordered map that takes joint name to joint index.
+std::unordered_map<std::string, size_t> get_namespaced_joint_map(const std::string& spot_name, bool has_arm) {
```

Comment:
@khughes-bdai meta: why have theses functions implemented in headers? They are not `inline`d nor anonymized using `static` or anonymous namespace, which means linkage issues when combining multiple shared libraries reusing these.

Same question for global constant variables.

Reply:
We threw a lot of stuff in a header when we were working on the dev branch just to get stuff working as quickly as possible and this is left over from that (not ideal). I'm planning to work on some reorganization of this package next that will include moving the implementation out of the header, just didn't want to get into it here for this bug fix